Hybrid solution documented in the charter

iherman opened the pull request:
I am mindful of the fact that the proposal must be published for a larger community soon (AC, plus other interested parties) to collect reactions. To make that comment period more efficient, I have:

- incorporated the alternative datatype of @pchampin into the main document
- I have also added a new approach to the document that I nicknamed 'hybrid' which outlines the approach we have been discussing in the separate issue https://github.com/w3c/rdf-dir-literal/issues/13
- I have prepared a version of the charter draft that offers a choice between the two versions: the pure `langString` update and the hybrid solution (I hope the way I did it in the text is understandable)

The readable version of the charter is: https://raw.githack.com/w3c/rdf-dir-literal/hybrid-solution/draft-charter.html

(The pr-preview works only on the core document...)

Cc: @pchampin @gkellogg @swickr @aphillips @chaals @dlongley @r12a @msporny (change proposals welcome!)


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-dir-literal/pull/14.html" title="Last updated on Jun 13, 2019, 6:13 PM UTC (ee97bf0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-dir-literal/14/89a8b8d...ee97bf0.html" title="Last updated on Jun 13, 2019, 6:13 PM UTC (ee97bf0)">Diff</a>

iherman committed:
Hybrid approach added

(That also included adding the new datatypes proposed by @pchampin)

iherman committed:
Updated the charter

it now includes the two versions as alternatives, with an explicit request for comments.

pchampin commented on index.html:
```diff
@@ -486,6 +535,46 @@ <h2>Pros and Cons</h2>
+
+            <ul>
+                <li>The core RDF 1.1 specification is updated, as described in <a href="#langString">the section above</a>.</li>
+                <li>New datatype(s) (as described <a href="datatype">above</a>) are defined, but with the following modifications:
```
```suggestion
                <li>New datatype(s) (as described <a href="#datatype">above</a>) are defined, but with the following modifications:
```

pchampin commented on index.html:
```diff
@@ -486,6 +535,46 @@ <h2>Pros and Cons</h2>
+            </p>
+
+            <p>
+                In view of their different usage pattersn, new versions of RDFa and JSON-LD may be updated, though, with the appropriate adaptations on the RDF Graphs they generate (e.g., if they generate N-Triples then the new datatypes should be used; if they are built on top of an RDF 1.2 compliant environment they would generate the proper value space entries).
```
```suggestion
                In view of their different usage patterns, new versions of RDFa and JSON-LD may be updated, though, with the appropriate adaptations on the RDF Graphs they generate (e.g., if they generate N-Triples then the new datatypes should be used; if they are built on top of an RDF 1.2 compliant environment they would generate the proper value space entries).
```

pchampin commented on index.html:
```diff
@@ -486,6 +535,46 @@ <h2>Pros and Cons</h2>
+                        <li>
+                            the value space for the datatype(s) consists of the renewed language string tuples as defined in the (renewed) RDF semantics document&nbsp;[[rdf11-mt]];
+                        </li>
+                        <li>the upcoming versions of N-Triple, N-Quads, Turtle, and TriG serializations are <em>not</em> modified; i.e., the only way to express the renewed <code>langString</code> is to use the new datatype(s).</li>
```
I was not suggesting that it would be the *only* way, only the *default* way. I would still like to be able to write in Turtle 1.2 or SPARQL 1.2  `"hello"@en^ltr` or may be `"hello"@en_ltr` (for the sake of regularity), instead of `"hello"^^i18n:en_ltr`, even though the latter would be the default for serializers (for backward compatibility).

Proposed rephrasing:

```suggestion
                        <li>in the upcoming versions of N-Triple, N-Quads, Turtle, and TriG, literals with a base direction would still be expressed using the <code>"value"^^i18n:XX_YY</code> syntax, but that syntax would be specially parsed as the corresponding (value, language-tag, direction) triple, <em>and the <code>rdf:langString</code> datatype</em>.</li>
```

For the sake of simpicity, I do not mention the possibility of other syntaxes, but I do not exclude it either.

iherman replied:
I formalized a bit differently, so check the latest version... but I believe I captured what you said (and agree with it!)

pchampin commented on index.html:
```diff
@@ -486,6 +535,46 @@ <h2>Pros and Cons</h2>
+                <li>New datatype(s) (as described <a href="datatype">above</a>) are defined, but with the following modifications:
+                    <ul>
+                        <li>
+                            these datatypes are defined to be deprecated soon, i.e., they are considered to be of a temporary usage only;
```
```suggestion
                            these datatypes are defined to be deprecated eventually, i.e., they are considered to be of a temporary usage only;
```

pchampin commented on index.html:
```diff
@@ -486,6 +535,46 @@ <h2>Pros and Cons</h2>
+            <p>
+                The major advantage of this approach is that it combines the ideal solution of taking care of an RDF “bug” while ensuring a somewhat smoother deployment.
+            </p>
+
```
```suggestion
            <p>
                A disadvantage is that it creates a special case in several concrete syntaxes (e.g. Turtle), where the `i18:XX_YY` family of IRIs are used as "magic" terms, and end up being interpreted as a different IRI (namely `rdf:langString`).
            </p>
```

pchampin commented on index.html:
```diff
@@ -486,6 +535,46 @@ <h2>Pros and Cons</h2>
+            </p>
+
+            <p>
+                The disadvantage is that there has to be a very careful consideration when the datatypes are deprecated, rescinded, etc., i.e., it will need a careful monitoring of the RDF ecosystem evolution.
```
```suggestion
                Aniother disadvantage is that there has to be a very careful consideration when the datatypes are deprecated, rescinded, etc., i.e., it will need a careful monitoring of the RDF ecosystem evolution.
```

pchampin commented on index.html:
```diff
@@ -201,68 +200,118 @@ <h2>Core definition</h2>
+                <section id="datatype-family">
+                    <h2>Define a <em>family</em> of language datatypes</h2>
+
+                    <p>A (very large, albeit finite) family of datatypes can be defined using the following URL patterns: <code>https://www.w3.org/i18n#XX{_YY}</code>, where:</p>
```
Bikeshedding alert. I would argue that the `i18n:` mapping should actually be mapped to an IRI derived on `<http://www.w3.org/1999/02/22-rdf-syntax-ns#langString>`, for example `<http://www.w3.org/1999/02/22-rdf-syntax-ns#langString#>`.

The rationale is the following: we are creating a strange situation in RDF 1.2, where
```sparql
  datatype("hello"^^i18:en_ltr) != i18n:en_ltr
```
but instead
```sparql
  datatype("hello"^^i18:en_ltr) == rdf:langString
```

It might be a little more acceptable if `i18:en_str` and `rdf:langString` were "related" IRIs, so to speak... Granted, this is a purely cosmetic and absolutely irrational argument. But sometimes cosmetics helps :)

iherman replied:
Yeah... except that what you propose is not a legal URL: the fragment ID is the last piece in a URL, and there can be only one... :-(

pchampin replied:
Oh, ok, I thought `#` was allowed in the `ifragment` part, but you are right, it is not...

But I'm fine changing it to `/` or `?` (which are allowed, I checked). That does not change the general idea of my proposal...

pchampin commented on index.html:
```diff
@@ -201,68 +200,118 @@ <h2>Core definition</h2>
+                        </dd>
+                        <dt>Value space</dt>
+                        <dd>
+                            Tuples consisting of the string, the `XX` tag extracted from the URL, and a directional tag, which is either the value of the `YY` or, if missing, the value of `auto`.
```
I didn't push the idea that far, but that's clever. It would mean that, semantically speaking, each `i18n:XX_YY` datatype is a subtype of the future `rdf:langString`... So the new interpretation in RDF 1.2 could be argued to be a built-in inference, rather than a dirty hack... I like that :-)

iherman replied:
Yeah, that is always what I had in mind: all that `i18N:*` is just a kind of syntactic trick with some serialization-level microsyntax on the URL, but would not affect the core of RDF and its semantics...

iherman committed:
Added PA's comments

iherman committed:
Added PA as editor

pchampin committed:
added my ORCID

also, removed my name from the acknowledgements; it is strange to acknowledge myself :)

pchampin committed:
a number of typos and minor rephrasing
